File: apps/backend/app/schemas/commission.py
# app/schema/commission.py

from datetime import datetime
from uuid import UUID
from typing import List
from pydantic import BaseModel, ConfigDict, Field
from app.schemas.transfer_method import TransferMethodRead
from app.schemas.user import UserBasicRead
from app.models.commission import CommissionWithdrawalStatus

class CommissionBalance(BaseModel):
    balance: float
    model_config = ConfigDict()

class CommissionTxRead(BaseModel):
    id: UUID
    type: str
    amount: float
    description: str | None
    created_at: datetime
    model_config = ConfigDict(from_attributes=True)

class PaginatedCommissionTx(BaseModel):
    total: int
    skip: int
    limit: int
    items: List[CommissionTxRead]
    model_config = ConfigDict()

class CommissionWithdrawalCreate(BaseModel):
    amount: float
    transfer_method_id: UUID

class CommissionWithdrawalRead(BaseModel):
    id: UUID
    amount: float
    status: CommissionWithdrawalStatus
    created_at: datetime = Field(..., alias="created_at")
    # dados do usuário que pediu o saque
    user: UserBasicRead
    # dados do PIX selecionado
    transfer_method: TransferMethodRead | None

    model_config = ConfigDict(from_attributes=True)

File: apps/portal-client/src/types/commission.ts
export interface CommissionBalance {
  balance: number;
}

export interface CommissionTxRead {
  id: string;
  type: string;
  amount: number;
  description?: string;
  created_at: string; // ISO date string
}

export interface PaginatedCommissionTx {
  total: number;
  skip: number;
  limit: number;
  items: CommissionTxRead[];
}

export interface CommissionWithdrawalCreate {
  amount: number;
  transfer_method_id: string;
}

export type CommissionWithdrawalStatus = 'PENDING' | 'COMPLETED' | 'FAILED';

export interface CommissionWithdrawalRead {
  id: string;
  amount: number;
  status: CommissionWithdrawalStatus;
  created_at: string; // ISO date string
  user: {
    id: string;
    name: string;
    email: string;
  };
  transfer_method?: {
    id: string;
    name: string;
    key_type: string;
    key_value: string;
    created_at: string;
  };
}
